### PR TITLE
Events: Never stop propagating CloseWidget

### PR DESCRIPTION
**Pull request by NiLuJe**
That seems like a rather terrible idea to begin with, and one that may actually have fatal consequences at that.

Re #7738

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7744)

### CODE REVIEW

**Comment by NiLuJe** [comment]
(Lightly tested).

@poire-z: Was some of this intended in InfoMessage/Notification?

(Most of the rest seem like someone copy-pasted the handlers from one widget that happened to do it this way).

**Comment by poire-z** [comment]
> Was some of this intended in InfoMessage/Notification?
(Most of the rest seem like someone copy-pasted the handlers from one widget that happened to do it this way).

I must admit that I haven't ever give this any thought (but I remember that you did recently) - so, I must have been one of these someones :)

But what's the use case ? Looks like `UIManager:close()` is the sole user, and does `widget:handleEvent(Event:new("CloseWidget"))` - so, there's only this `widget` that will get it.
ReaderUI/FM may forward it to all their ReaderSomething and FileManagerSomething children widgets - but I'm not sure all the ones you removed `return true` (mostly independant top UI widgets drawn over Reader/FM) will get it from ReaderUI, will they ? So, their return values feels like it shouldn't matter ?

File: frontend/ui/uimanager.lua [hunk 1]
```diff
@@ -467,16 +467,16 @@ function UIManager:close(widget, refreshtype, refreshregion, refreshdither)
...
     logger.dbg("close widget:", widget.name or widget.id or tostring(widget))
     local dirty = false
-    -- Ensure all the widgets can get onFlushSettings event.
+    -- First notify the closed widget to save its settings...
     widget:handleEvent(Event:new("FlushSettings"))
-    -- first send close event to widget
+    -- ...and notify it that it ought to be gone now.
     widget:handleEvent(Event:new("CloseWidget"))
-    -- make it disabled by default and check if any widget wants it disabled or enabled
+    -- Make it disabled by default and check if any widget wants it disabled or enabled
```
Comment (Frenzie) [inline frontend/ui/uimanager.lua:474]: If you're going to change the line, change the line. :-P
```suggestion
    -- Make it's disabled by default and check if there are any widgets that want it disabled or enabled.
```

File: frontend/ui/uimanager.lua [hunk 2]
```diff
@@ -467,16 +467,16 @@ function UIManager:close(widget, refreshtype, refreshregion, refreshdither)
...
     Input.disable_double_tap = true
     local requested_disable_double_tap = nil
     local is_covered = false
     local start_idx = 1
-    -- then remove all references to that widget on stack and refresh
+    -- Then remove all references to that widget on stack and refresh
```
Comment (Frenzie) [inline frontend/ui/uimanager.lua:479]:
```suggestion
    -- Then remove all references to that widget on stack and refresh.
```

**Review by Frenzie — COMMENTED** [review]
What exactly is the problem you're trying to fix?

**Comment by NiLuJe** [comment]
One module of both ReaderUI and FileManager appears to be stopping the event along the way, which means it never actually reaches them (events go children first), which breaks the instance tracking.

**Comment by NiLuJe** [comment]
> > Was some of this intended in InfoMessage/Notification?
> > (Most of the rest seem like someone copy-pasted the handlers from one widget that happened to do it this way).
> 
> I must admit that I haven't ever give this any thought (but I remember that you did recently) - so, I must have been one of these someones :)
> 
> But what's the use case ? Looks like `UIManager:close()` is the sole user, and does `widget:handleEvent(Event:new("CloseWidget"))` - so, there's only this `widget` that will get it.
> ReaderUI/FM may forward it to all their ReaderSomething and FileManagerSomething children widgets - but I'm not sure all the ones you removed `return true` (mostly independant top UI widgets drawn over Reader/FM) will get it from ReaderUI, will they ? So, their return values feels like it shouldn't matter ?

Yep. Pretty much clutching at straws here ;).

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 19 of 20 files at r1, 1 of 1 files at r2.
> Dismissed @Frenzie from 2 discussions.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/7744)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @NiLuJe)

___
*[frontend/ui/uimanager.lua, line 474 at r1](https://reviewable.io/reviews/koreader/koreader/7744#-MaPMekIkGqLfIfRKvhe-r1-474:-MaV8c46DPiY5-2Xq_VZ:b-896fix) ([raw file](https://github.com/koreader/koreader/blob/49c249aa00d3d4de1d2b10462be47ae4c12988a5/frontend/ui/uimanager.lua#L474)):*
<details><summary><i>Previously, Frenzie (Frans de Jonge) wrote…</i></summary><blockquote>

If you're going to change the line, change the line. :-P
```suggestion
    -- Make sure it's disabled by default and check if there are any widgets that want it disabled or enabled.
```
</blockquote></details>

Done.
___
*[frontend/ui/uimanager.lua, line 479 at r1](https://reviewable.io/reviews/koreader/koreader/7744#-MaPMekIkGqLfIfRKvhe-r1-479:-MaV8eXIBM8NlTMXgVwA:b-896fix) ([raw file](https://github.com/koreader/koreader/blob/49c249aa00d3d4de1d2b10462be47ae4c12988a5/frontend/ui/uimanager.lua#L479)):*
<details><summary><i>Previously, Frenzie (Frans de Jonge) wrote…</i></summary><blockquote>

```suggestion
    -- Then remove all references to that widget on stack and refresh.
```
</blockquote></details>

Done.

**Comment by NiLuJe** [comment]
Whelp. If for no other reason than documentation purposes... ;).

But, technically, can be useful for Widgets to use as a cheap finalizer ;).

**Pull request merged**